packages/gatsby/src/query-runner/query-compiler.js

```diff
@@ -50,7 +50,8 @@ class Runner {
   }
 
   async parseEverything() {
-    let files = await globp(`${this.baseDir}/**/*.js`)
+    let files = await globp(`${this.baseDir}/**/*.+(t|j)s?(x)`)
+    files = files.filter(d => !d.match(/(\.stories\.|\.test\.|\.d\.)/))
```

Comment:
All this files are in src (one folder for one module, with tests beside component) 
My only issue is with d.ts (needed for js module and global var as graphql) I can remove test and stories if you want ;) It was just to avoid useless processing